karapace/in_memory_database.py

```diff
@@ -184,6 +184,17 @@ def find_schemas(self, *, include_deleted: bool, latest_only: bool) -> dict[Subj
                 res_schemas[subject] = selected_schemas
         return res_schemas
 
+    def subjects_for_schema(self, schema_id: SchemaId) -> list[Subject]:
+        subjects = []
+        with self.schema_lock_thread:
+            for subject, subject_data in self.subjects.items():
+                for version in subject_data.schemas.values():
+                    if version.deleted is False and version.schema_id == schema_id:
+                        subjects.append(subject)
+                        break
```

Comment:
question: Why this break? Is a schema guaranteed to have only one subject, if so, why return a list?

Reply:
The break it's for the iteration over the values of the map for the subject, not for the iteration over all the subjects. It's like: if you found this schema for this subject there's no reason to iterate over the other schemas, just add it to the list and start for the next subject

Reply:
Right, I shouldn't be reading code before having coffee 🙈